lib/vfscore/automount.c

```diff
@@ -79,17 +80,26 @@ struct vfscore_volume {
 };
 
 #if CONFIG_LIBUKCPIO && CONFIG_LIBRAMFS
-static int do_mount_initrd(const void *initrd, size_t len, const char *path)
+static int do_mount_initrd(const void *initrd, size_t len, const char *path,
+			   unsigned long flags, const char *opts)
 {
 	int rc;
 
 	UK_ASSERT(path);
 
-	rc = mount("", path, "ramfs", 0x0, NULL);
-	if (unlikely(rc)) {
-		uk_pr_crit("Failed to mount ramfs to \"%s\": %d\n",
-			   path, errno);
-		return -1;
+	/* 'extract' is an (e)initrd-specific mount option that allows
+	 * extracting an archive to an existing filesystem. If this option
+	 * is chosen, no other mount option is accepted.
+	 */
+	if (!opts || (strcmp(opts, LIBVFSCORE_INITRD_OPT_EXTRACT) != 0)) {
```

Comment:
Would it make sense to introduce an option macro for `mkmp` as well?

Reply:
I added an issue for that, in order to keep this PR simple: #1229